### ISSUE TITLE
SLF001 rejects acceptable private and protected access

### ISSUE THREAD

**Issue by sanmai-NL** [charliermarsh/ruff#2893]
In the following case, a static method is called from within the class definition. 
That should be allowed, and from subclasses too.

Calls to private methods should also be allowed from within the class definition.

```python
class TestSLF001:
    @staticmethod
    def _protected() -> None:
        pass

    def public() -> None:
        TestSLF001._protected()
```

```sh
ruff check --select ALL --ignore D103,TCH003,D testslf001.py
```

ruff 0.0.245

`pyproject.toml` snippet:

```toml
[tool.ruff]
select = ["ALL"]
line-length = 120
target-version = "py311"
```

**Comment by charliermarsh** [comment]
👍 Yeah we should accept this. (It currently accepts `self._private` and `cls._private`, if it's helpful in the interim.)

**Comment by sanmai-NL** [comment]
`cls` not being the first param of the static method warrants a linting warning in its own right, indeed.
